Sources/CoreAardvark/Logging/ARKDataArchive.m

```diff
@@ -132,18 +132,8 @@ - (void)readObjectsFromArchiveWithCompletionHandler:(nonnull void (^)(NSArray *
                     // We're done.
                     break;
                 }
-                
-                id object = nil;
-                @try {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated"
-                    object = [NSKeyedUnarchiver unarchiveObjectWithData:objectData];
-#pragma clang diagnostic pop
-                }
-                @catch (NSException *exception) {
-                    // The structure of the archive itself isn't corrupted, just the data itself, so ignore and continue.
-                    continue;
-                }
+
+                id object = [NSKeyedUnarchiver unarchivedObjectOfClass:objectType fromData:objectData error:NULL];
```

Comment:
Previously we'd skip the work below if there was an error (see the `continue` in the `@catch` above). We're removing that functionality in this PR. Is that intentional?

Reply:
Per [`-[NSKeyedUnarchiver unarchivedObjectOfClass:fromData:error:]`](https://developer.apple.com/documentation/foundation/nskeyedunarchiver/2962884-unarchivedobjectofclass) documentation, this returns:

> The decoded root of the object graph, or nil if an error occurred.

So if there's an error (what would previously have thrown an exception), we should get nil back here, which will then fail the check below and skip adding it to the array anyway. Seems unnecessary to explicitly check for an error as well since it wouldn't change the outcome.

Reply:
ahh that makes sense. I thought we'd need to inspect the `error`. We good!